tests/benchmarks/test_work_stealing.py

```diff
@@ -0,0 +1,111 @@
+import time
+
+import dask.array as da
+import distributed
+import numpy as np
+import pytest
+from coiled.v2 import Cluster
+from dask import delayed, utils
+from distributed import Client, wait
+from tornado.ioloop import PeriodicCallback
+
+
+@pytest.mark.xfail(reason="https://github.com/dask/distributed/issues/6573")
+def test_trivial_workload_should_not_cause_work_stealing(small_client):
+    root = delayed(lambda n: "x" * n)(utils.parse_bytes("1MiB"), dask_key_name="root")
+    results = [delayed(lambda *args: None)(root, i) for i in range(10000)]
+    futs = small_client.compute(results)
+    wait(futs)
+    _ = [fut.result() for fut in futs]
```

Comment:
```suggestion
    client.gather(futs)
```

will give the same result